devicetypes/vision-raytseng/vision-4-in-1-motion-sensor.src/vision-4-in-1-motion-sensor.groovy

```diff
@@ -0,0 +1,495 @@
+			if (config?."${getConfigurationInfo(idx, "name")}" == state."${getConfigurationInfo(idx, "name")}") {
+				if (state."${getConfigurationInfo(idx, "name")}Refresh" == true) {
+					state."${getConfigurationInfo(idx, "name")}Refresh" = false
+				}
+			}
+		}
+	}
+}
+
+def getConfigurationNumber() {
+	return [1, 2, 3, 4, 5, 6, 7, 8]
+}
+
+def getConfigurationInfo(num, text) {
+	def parameter = [:]
+	
+	parameter.parameter1name = "TemperatureUnit"
+	parameter.parameter1title = "Temperature Unit [°C/°F]"
+	parameter.parameter1description = ""
+	parameter.parameter1type = "enum"
+	parameter.parameter1options = ["°C", "°F"]
+	parameter.parameter1enumMap = ["°C": 0, "°F": 1]
+	parameter.parameter1default = "°C"
+	parameter.parameter1size = 1
+	
+	parameter.parameter2name = "TempReportWhenChanged"
+	parameter.parameter2title = "Report when temperature difference is over the setting [unit is 0.1°C/°F]"
+	parameter.parameter2description = ""
+	parameter.parameter2type = "number"
+	parameter.parameter2options = "1..50"
+	parameter.parameter2enumMap = []
+	parameter.parameter2default = 30
+	parameter.parameter2size = 1
+	
+	parameter.parameter3name = "HumiReportWhenChanged"
+	parameter.parameter3title = "Report when humidity difference is over the setting [%]"
+	parameter.parameter3description = ""
+	parameter.parameter3type = "number"
+	parameter.parameter3options = "1..50"
+	parameter.parameter3enumMap = []
+	parameter.parameter3default = 20
+	parameter.parameter3size = 1
+	
+	parameter.parameter4name = "LightReportWhenChanged"
+	parameter.parameter4title = "Report when illuminance difference is over the setting [%](1% is approximately equal to 4.5 lux)"
+	parameter.parameter4description = ""
+	parameter.parameter4type = "number"
+	parameter.parameter4options = "5..50"
+	parameter.parameter4enumMap = []
+	parameter.parameter4default = 25
+	parameter.parameter4size = 1
+	
+	parameter.parameter5name = "MotionRestoreTime"
+	parameter.parameter5title = "Motion inactive report time [Minutes] after active"
+	parameter.parameter5description = ""
+	parameter.parameter5type = "number"
+	parameter.parameter5options = "1..127"
+	parameter.parameter5enumMap = []
+	parameter.parameter5default = 3
+	parameter.parameter5size = 1
+	
+	parameter.parameter6name = "MotionSensitivity"
+	parameter.parameter6title = "Motion active sensitivity"
+	parameter.parameter6description = ""
+	parameter.parameter6type = "enum"
+	parameter.parameter6options = ["Highest", "Higher", "High", "Medium", "Low", "Lower", "Lowest"]
+	parameter.parameter6enumMap = ["Highest": 1, "Higher": 2, "High": 3, "Medium": 4, "Low": 5, "Lower": 6, "Lowest": 7]
+	parameter.parameter6default = "Medium"
+	parameter.parameter6size = 1
+	
+	parameter.parameter7name = "LedDispMode"
+	parameter.parameter7title = "LED display mode"
+	parameter.parameter7description = ""
+	parameter.parameter7type = "enum"
+	parameter.parameter7options = ["LED off when Temperature report/Motion active",
```

Comment:
Please, could You fix indentations in parameter7options and parameter7enumMap ?

Reply:
Okay, I will fix the indentations in parameter7options and parameter7enumMap.

Reply:
Looks better now, but now spacing has to be fixed (replace 4 spaces to 1 tab).

Reply:
These was used 8 tabs and 3 spaces to make these indentations originally before I fixed today, but it looks not alignment. So I replace the 7 tabs to 28 spaces to fix this look. It will be the same as previous committed If I replace 4 spaces to 1 tab.

Reply:
Since indentation in this file contains tabs - please use tabs in this case too (it doesn't have to be super aligned with the first element)

Reply:
So these indentations are allowed?
```
parameter.parameter7options = ["LED off when Temperature report/Motion active",
		"LED blink when Temperature report/Motion active",
		"LED blink when Motion active/LED off when Temperature report"]
	parameter.parameter7enumMap = ["LED off when Temperature report/Motion active": 1,
		"LED blink when Temperature report/Motion active": 2,
		"LED blink when Motion active/LED off when Temperature report": 3]
```

Reply:
Yes, it would look better this way.

Reply:
I got it, thank you.